pkg/security/scc/byrestrictions.go

```diff
@@ -111,3 +126,41 @@ func volumePointValue(scc *securityapi.SecurityContextConstraints) points {
 	}
 	return noPoints
 }
+
+// hasCap checks for needle in haystack.
+func hasCap(needle string, haystack []kapi.Capability) bool {
+	for _, c := range haystack {
+		if needle == strings.ToUpper(string(c)) {
+			return true
+		}
+	}
+	return false
+}
+
+// capabilitiesPointValue returns a score based on the capabilities allowed,
+// added, or removed by the SCC. This allow us to prefer the more restrictive
+// SCC.
+func capabilitiesPointValue(scc *securityapi.SecurityContextConstraints) points {
+	capsPoints := capDefaultPoints
+	capsPoints += capAddOnePoints * points(len(scc.DefaultAddCapabilities))
+	if hasCap(string(kapi.CapabilityAll), scc.AllowedCapabilities) {
+		capsPoints += capAllowAllPoints
+	} else if hasCap("ALL", scc.AllowedCapabilities) {
+		capsPoints += capAllowAllPoints
+	} else {
+		capsPoints += capAllowOnePoints * points(len(scc.AllowedCapabilities))
+	}
+	if hasCap(string(kapi.CapabilityAll), scc.RequiredDropCapabilities) {
```

Comment:
It might not support it but it does not prevent it either, it allows it and creates the SCC with it ...